### DISCUSSION TITLE
Offload tasks using a SCSI interface IC

### DISCUSSION THREAD

**Discussion by bobdvb** [erichelgeson/BlueSCSI-v1#266] [Ideas]
I've noticed that neither the BlueSCSI nor the other recent emulators use a SCSI interface IC. These ICs take some of the SCSI tasks and embed them into the chip, requiring less processing and supporting DMA transfers. Chips like the NEC 53C9x or Qlogic FAS2x6/FAS368M series for example.

I understand that it's likely people would want to avoid chips that aren't made anymore, but many of them were so widely produced that large quantities of them are still out there. There are also multiple vendors for the NEC chip (including AMD). So, supply is not limited particularly, and they're relatively cheap. Sometimes cheaper than the level shifters needed in direct bus access designs.

I've seen one other project which does this, but it hasn't had much attention (tinyscsiemulator). Perhaps there's an opportunity to improve performance and simplify the design?

**Comment by erichelgeson** [comment]
You hit the nail on the head - discontinued chips. Since BlueSCSI v2 can hit ~10mb/sec and cost to build them yourself is still extremely low - the level of effort doesn't seem worth it. But it could still be a fun project to explore.
